src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlFormatReaderGenerator.cs

```diff
@@ -98,17 +98,19 @@ private class CriticalHelper
             private ArgBuilder _memberNamesArg;
             private ArgBuilder _memberNamespacesArg;
             private ArgBuilder _collectionContractArg;
-
-            private bool _useReflection = true;
+#endif
 
             public XmlFormatClassReaderDelegate GenerateClassReader(ClassDataContract classContract)
             {
-                if (_useReflection)
+                if (DataContractSerializer.Option == SerializationOption.ReflectionAsBackup || DataContractSerializer.Option == SerializationOption.ReflectionOnly)
                 {
```

Comment:
Does this also work for CoreCLR?

---
In reply to: [63755899](https://github.com/dotnet/corefx/pull/8629#discussion_r63755899) [](ancestors = 63755899)